### PR TITLE
feat: show top banner for broken integrations [Cal 3125]

### LINKED ISSUE #13667: [CAL-3125] show top banner for broken integrations
when we show this:

image

we should also show the top banner and link to the app that needs to be reinstalled

CleanShot 2024-02-13 at 22 24 52@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3125](https://linear.app/calcom/issue/CAL-3125/show-top-banner-for-broken-integrations)</sub>

### PR DESCRIPTION
**Pull request by Hitesh-Sisara**
## What does this PR do?
 check for invalid credentials and display the banner 

video : https://www.loom.com/share/aa0ed6ad42ea422485ce068ae39e0c9f?sid=9dd3a6f4-79f7-4ed5-b143-2de52ad0ab21

Fixes #13667
/claim #13667

## Type of change
- New feature (non-breaking change which adds functionality)

- My code follow the style guidelines of this project
- I  checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/features/users/components/InvalidAppCredentialsBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export type InvalidAppCredentialBannersProps = {
+  data: RouterOutputs["viewer"]["getUserTopBanners"]["invalidAppCredentialBanners"];
+};
+
+export function InvalidAppCredentialBanners({ data }: InvalidAppCredentialBannersProps) {
+  if (data.length === 0) {
+    return null; // No need to show banner if the array is empty
+  }
+
+  console.log("banner data: ", data);
```
Comment (Udit-takkar) [inline packages/features/users/components/InvalidAppCredentialsBanner.tsx:16]: remove log

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -1599,7 +1599,8 @@
   "attendee_email_variable": "Attendee email",
   "attendee_email_info": "The person booking's email",
   "kbar_search_placeholder": "Type a command or search...",
-  "invalid_credential": "Oh no! Looks like permission expired or was revoked. Please reinstall again.",
+  "invalid_credential": "looks like permissions/credentials expired or were revoked for",
```
Comment (Udit-takkar) [inline apps/web/public/static/locales/en/common.json:1602]: This should be "It looks like permissions expired or were revoked for {{appName}}."

**Review by Udit-takkar — COMMENTED** [review]
@Hitesh-Sisara The implementation looks good but I am not sure if we are updating invalid column to true for all apps when they fail.

File: packages/trpc/server/routers/loggedInViewer/checkForInvalidAppCredentials.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    select: {
+      app: {
+        select: {
+          slug: true,
+        },
+      },
+    },
+  });
+
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/loggedInViewer/checkForInvalidAppCredentials.ts:26]: we might also wanna check team credentials that the user is admin/owner of

Comment (Hitesh-Sisara) [reply packages/trpc/server/routers/loggedInViewer/checkForInvalidAppCredentials.ts:26]: ok

**Comment by Hitesh-Sisara** [comment]
@Udit-takkar made requested changes have a look

**Review by Udit-takkar — COMMENTED** [review]
LGTM. code looks fine

**Comment by ChaudharyPradip** [comment]
Hey @Udit-takkar,
In the file packages/features/shell/Shell.tsx on line 271 why are we using if else ladder when we do the same thing in each case?
If we change it from using if else to normal return then it will make the code smaller as well as reduce 1 step for the next contributor

**Comment by Udit-takkar** [comment]
> Hey @Udit-takkar, In the file packages/features/shell/Shell.tsx on line 271 why are we using if else ladder when we do the same thing in each case? If we change it from using if else to normal return then it will make the code smaller as well as reduce 1 step for the next contributor

I was getting type error when i didn't use if else statements

**Comment by Hitesh-Sisara** [comment]
@Udit-takkar @ChaudharyPradip can you merge ?

**Comment by ChaudharyPradip** [comment]
Hey @Hitesh-Sisara, I am not a maintainer

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**